src/Polysemy/Async.hs

```diff
@@ -57,9 +69,37 @@ asyncToIO m = withLowerToIO $ \lower _ -> lower $
     )  m
 {-# INLINE asyncToIO #-}
 
+------------------------------------------------------------------------------
+-- | Run an 'Async' effect in terms of 'A.async' through final 'IO'.
+--
+-- /Beware/: Effects that aren't interpreted in terms of 'IO'
+-- will have local state semantics in regards to 'Async' effects
+-- interpreted this way. See 'Final'.
+--
+-- Notably, unlike 'asyncToIO', this is not consistent with
+-- 'Polysemy.State.State' unless 'Polysemy.State.runStateIORef' is used.
+-- State that seems like it should be threaded globally throughout 'Async'
+-- /will not be./
+--
+-- Use 'asyncToIO' instead if you need to run
+-- pure, stateful interpreters after the interpreter for 'Async'.
+-- (Pure interpreters are interpreters that aren't expressed in terms of
+-- another effect or monad; for example, 'Polysemy.State.runState'.)
```

Comment:
Is this any better? It's the wording from `polysemy-zoo`, but with clarification on what is meant by "pure".

Reply:
Good!